nemo/collections/multimodal/speech_llm/data/build_dataset.py

```diff
@@ -0,0 +1,233 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import copy
+from pathlib import Path
+
+import torch
+from megatron.core import parallel_state
+from omegaconf.omegaconf import OmegaConf
+
+from nemo.collections.asr.parts.preprocessing.perturb import process_augmentations
+from nemo.collections.multimodal.speech_llm.data.audio_text_dataset import (
+    get_audio_text_dataset_from_config,
+    get_tarred_audio_text_dataset_from_config,
+)
+from nemo.collections.multimodal.speech_llm.data.lhotse_dataset import LhotseAudioQuestionAnswerDataset, TextProcessing
+from nemo.collections.nlp.data.language_modeling.megatron.blendable_dataset import BlendableDataset
+from nemo.collections.nlp.data.language_modeling.megatron.megatron_batch_samplers import (
+    MegatronPretrainingBatchSampler,
+)
+from nemo.utils import logging
+
+
+def build_salm_dataset(model_instance, data_cfg, is_train):
```

Comment:
This is a function for creating dataset more conveniently. Try to name this as salm style since we already have salm in conf/. I imagine we will add build_.*_dataset for other styles in future

Reply:
The 'salm' in config is referring to a specific set of params we use in the SALM paper, not referring to a general arch. Same for 'bestow'. we should use more general names than model-specific names, just like we're trying to avoid having 'canary' in the code except for the prompt format

Reply:
DOne